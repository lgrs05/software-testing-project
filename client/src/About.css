
.About {
  font-family: Cochin;
  text-align: center;
}


h1{
  margin-top: -580px;
  z-index: 100;
}

section{
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.about1{
  margin-left: auto;
  margin-right: auto;
  clear: both;
  padding: 10px;
  display:flex;
  align-items: center;
  padding-left: 400px;
  font-size: 40px;
  text-indent: 30px;
}



.about2{
  margin-left: auto;
  margin-right: auto;
  clear: both;
  padding: 10px;
  display:flex;
  align-items: center;
  padding-left: 400px;
  font-size: 40px;
  text-indent: 30px;
}

.about3{
  margin-left: auto;
  margin-right: auto;
  clear: both;
  padding: 10px;
  display:flex;
  align-items: center;
  padding-left: 400px;
  font-size: 40px;
  text-indent: 30px;
}