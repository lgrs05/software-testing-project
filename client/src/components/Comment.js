import Avatar from "@material-ui/core/Avatar";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
import PropTypes from 'prop-types';
import React from "react";
import Typography from "@material-ui/core/Typography";

const avatarSrc = "https://source.unsplash.com/2ShvY8Lf6l0/800x599";

function Comment({name, message, time}) {
    return (
        <ListItem alignItems="flex-start">
            <ListItemAvatar>
                <Avatar alt={name} src= {avatarSrc} />
            </ListItemAvatar>
            <ListItemText
                primary={name}
                secondary={
                    <React.Fragment>
                        <Typography component="span" inline color="textPrimary">
                            {message}
                        </Typography>
                        <br />{time}
                    </React.Fragment>
                }
            />
        </ListItem>
    );
}


Comment.propTypes = {
    name    : PropTypes.string.isRequired,
    message : PropTypes.string.isRequired,
    time    : PropTypes.string.isRequired
};


export default (Comment);