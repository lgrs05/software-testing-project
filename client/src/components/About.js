import '../About.css';
import About1 from "../circleabout1.png";
import About2 from "../circleabout2.png";
import About3 from "../circleabout5.png";
import Background from "../background4.jpg";
import PropTypes from 'prop-types';
import React from "react";
import {withStyles} from '@material-ui/core/styles';

const sectionStyle = {
    width           : "100%",
    height          : "650px",
    backgroundSize  : "auto",
    backgroundImage : "url(" +  Background  + ")"
};

const image1 = {
    width           : "100px",
    height          : "100px",
    backgroundImage : "url(" + About1 + ")"
};

const image2 = {
    width           : "100px",
    height          : "100px",
    backgroundImage : "url(" + About2 + ")"
};

const image3 = {
    width           : "100px",
    height          : "100px",
    backgroundImage : "url(" + About3 + ")"
};

const styles = (theme) => ({
    paper : {
        position        : 'absolute',
        width           : theme.spacing.unit * 50,
        backgroundColor : theme.palette.background.paper,
        boxShadow       : theme.shadows[5],
        padding         : theme.spacing.unit * 4,
        outline         : 'none'
    },
    toolbarButton : {
        marginLeft  : -12,
        marginRight : 20
    },
    button : {
        margin : theme.spacing.unit
    }
});

function About(){
    return (<div className="About">
        <section style={ sectionStyle }>
        </section>
        <h1>About</h1>
        <h2>Start sharing moments with Slika Share today! On</h2>
        <h2>Slika Share you can upload and post photos, comment</h2>
        <h2>on friends photos and even share them.</h2>
        <div className={"about1"}>
            <img src={About1} alt={image1} />
            <span>Post Moments</span>
        </div>
        <div className={"about3"}>
            <img src={About3} alt={image3}  />
            <span>Share Moments</span>
        </div>
        <div className={"about2"}>
            <img src={About2} alt={image2}  />
            <span>Say something</span>
        </div>

    </div>
    );

}

About.propTypes = {
    classes : PropTypes.object.isRequired
};

export default withStyles(styles)(About);
