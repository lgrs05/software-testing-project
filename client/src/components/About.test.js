import About from './About'
import Comment from './Comment';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

let comm = {};

beforeEach(() => {
    comm = {
        name    : 'Pedro',
        message : "This is a good message",
        time    : new Date(2019, 2, 21, 12, 0, 0, 0).toLocaleString()
    };
});

it('renders correctly', () => {
    const tree = renderer.create(<Comment
        name={comm.name}
        message={comm.message}
        time={comm.time} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('About renders correctly', () => {
    const tree = renderer.create(<About />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
