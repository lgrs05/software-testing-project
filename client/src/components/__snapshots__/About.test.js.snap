// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<div
  className="About"
>
  <section
    style={
      Object {
        "backgroundImage": "url(background4.jpg)",
        "backgroundSize": "auto",
        "height": "650px",
        "width": "100%",
      }
    }
  />
  <h1>
    About
  </h1>
  <h2>
    Start sharing moments with Slika Share today! On
  </h2>
  <h2>
    Slika Share you can upload and post photos, comment
  </h2>
  <h2>
    on friends photos and even share them.
  </h2>
  <div
    className="about1"
  >
    <img
      alt={
        Object {
          "backgroundImage": "url(circleabout1.png)",
          "height": "100px",
          "width": "100px",
        }
      }
      src="circleabout1.png"
    />
    <span>
      Post Moments
    </span>
  </div>
  <div
    className="about3"
  >
    <img
      alt={
        Object {
          "backgroundImage": "url(circleabout5.png)",
          "height": "100px",
          "width": "100px",
        }
      }
      src="circleabout5.png"
    />
    <span>
      Share Moments
    </span>
  </div>
  <div
    className="about2"
  >
    <img
      alt={
        Object {
          "backgroundImage": "url(circleabout2.png)",
          "height": "100px",
          "width": "100px",
        }
      }
      src="circleabout2.png"
    />
    <span>
      Say something
    </span>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<li
  className="MuiListItem-root-1 MuiListItem-default-4 MuiListItem-dense-5 MuiListItem-gutters-9 MuiListItem-alignItemsFlexStart-6"
  disabled={false}
>
  <div
    className="MuiAvatar-root-16 MuiListItemAvatar-alignItemsFlexStart-14"
  >
    <img
      alt="Pedro"
      className="MuiAvatar-img-18"
      src="https://source.unsplash.com/2ShvY8Lf6l0/800x599"
    />
  </div>
  <div
    className="MuiListItemText-root-19"
  >
    <span
      className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-primary-22"
    >
      Pedro
    </span>
    <p
      className="MuiTypography-root-25 MuiTypography-body1-34 MuiTypography-colorTextSecondary-58 MuiListItemText-secondary-23"
    >
      <span
        className="MuiTypography-root-25 MuiTypography-body1-34 MuiTypography-colorTextPrimary-57 MuiTypography-inline-60"
      >
        This is a good message
      </span>
      <br />
      2019-3-21 12:00:00
    </p>
  </div>
</li>
`;
